tcpdevmem.c

```diff
@@ -0,0 +1,119 @@
+#include <linux/if.h>
+#include <linux/dma-buf.h>
+#include <sys/ioctl.h>
+
+#include "flow.h"
+#include "lib.h"
+#include "logging.h"
+#include "tcpdevmem_cuda.h"
+#include "tcpdevmem.h"
+#include "thread.h"
+
+#define TEST_PREFIX "ncdevmem"
+#define RETURN_IF_NON_ZERO(cmd)	\
+	ret = (cmd);		\
+	if (ret) return ret;
+
+int driver_reset(const struct options *opts) {
+	char driver_reset_cmd[512];
+	int ret = 0;
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-strict-header-split on", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-strict-header-split off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-header-split off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-rxfh-indir %s equal 16", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool -K %s ntuple off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-strict-header-split off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-header-split off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool -K %s ntuple off", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool --set-priv-flags %s enable-max-rx-buffer-size on", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	sprintf(driver_reset_cmd, "ethtool -K %s ntuple on", opts->tcpd_link_name);
+	RETURN_IF_NON_ZERO(system(driver_reset_cmd));
+
+	return ret;
+}
+
+int install_flow_steering(const struct options *opts, intptr_t buf,
+			  struct thread *t)
+{
+	int q_start = opts->queue_start;
+	int q_num = opts->queue_num;
+	int ret;
+
+	int num_queues = q_start + (t->index % q_num);
+	printf("Bind to queue %i\n", num_queues);
```

Comment:
remove debug looking printf?

If you want, maybe use something like LOG_DEBUG.

Reply:
I think it's useful to have some prints to stdout (rather than logging) to convey a sense of progress to the user looking at the CLI